programs/editor/Tools.js

```diff
@@ -62,24 +62,32 @@ define("webodf/editor/Tools", [
                 aboutDialog,
                 sessionSubscribers = [];
 
+            function placeAndStartUpWidget(widget) {
+                widget.placeAt(toolbar);
+                widget.startup();
+            }
+
             /**
              * Creates a tool and installs it, if the enabled flag is set to true.
              * Only supports tool classes whose constructor has a single argument which
              * is a callback to pass the created widget object to.
              * @param {!function(new:Object, function(!Object):undefined)} Tool  constructor method of the tool
              * @param {!boolean} enabled
+             * @param {!Object|undefined=} config
```

Comment:
Given each tool might have a completely different config data structure possibly not. The config structure is passed here as a blob, so proper type checking has to be done at other places in any case.